### PR TITLE
feat(invitation): add email notification for invitees

### PR DESCRIPTION
**Pull request by devformatters**

### CODE REVIEW

**Review by guoliu — APPROVED** [review]

File: src/mutations/circle/invite.ts [hunk 1]
```diff
@@ -122,9 +147,63 @@ const resolver: MutationToInviteResolver = async (
...
+  for (const invitation of invitations) {
+    let code
+    let recipient
+    let redirectUrl
+    const { email, userId } = invitation
+
+    if (userId) {
+      recipient = await atomService.findFirst({
+        table: 'user',
+        where: { id: userId },
```
Comment (robertu7) [inline src/mutations/circle/invite.ts:160]: active user only?

File: src/mutations/circle/invite.ts [hunk 2]
```diff
@@ -122,9 +147,63 @@ const resolver: MutationToInviteResolver = async (
...
+      })
+    } else {
+      recipient = await atomService.findFirst({
+        table: 'user',
+        where: { email },
```
Comment (robertu7) [inline src/mutations/circle/invite.ts:165]: active user only?

**Review by robertu7 — APPROVED** [review]

File: src/mutations/circle/invite.ts (hunk 1 above)
Comment (devformatters) [reply src/mutations/circle/invite.ts]: Added.

File: src/mutations/circle/invite.ts (hunk 2 above)
Comment (devformatters) [reply src/mutations/circle/invite.ts]: Added.

File: src/mutations/circle/invite.ts [hunk 3]
```diff
@@ -122,9 +147,63 @@ const resolver: MutationToInviteResolver = async (
...
+      })
+    }
+
+    // if user not found by id and email, then generate code
+    if (!recipient && email) {
+      code = await userService.createVerificationCode({
+        email,
+        type: VERIFICATION_CODE_TYPES.register,
+        strong: true,
```
[the remark this answers is not in the document]
Comment (devformatters) [reply src/mutations/circle/invite.ts:179]: Changed.

**Pull request merged**